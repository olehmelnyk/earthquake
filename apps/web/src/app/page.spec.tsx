import { render, screen } from '@testing-library/react';
import Page from './page';
import { describe, it, expect, vi } from 'vitest';

// Mock the Dashboard component (we've already tested it)
vi.mock('./components/Dashboard', () => ({
  Dashboard: () => <div data-testid="dashboard">Dashboard Component</div>
}));

describe('Page', () => {
  it('should render the Dashboard component', () => {
    render(<Page />);

    expect(screen.getByTestId('dashboard')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<Page />);
    expect(container).toMatchSnapshot();
  });
});