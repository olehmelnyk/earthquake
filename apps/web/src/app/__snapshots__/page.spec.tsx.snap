// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page > should match snapshot 1`] = `
<div>
  <div
    data-testid="dashboard"
  >
    Dashboard Component
  </div>
</div>
`;
